internal/osutils/lockfile/pidlock.go

```diff
@@ -94,6 +95,27 @@ func (pl *PidLock) Close(keepFile ...bool) error {
 	return nil
 }
 
+// WaitForLock will attempt to acquire the lock for the duration given
+func (pl *PidLock) WaitForLock(timeout time.Duration) error {
+	timer := time.NewTimer(timeout)
+	for {
+		_, err := pl.TryLock()
+		if err != nil {
+			if !errs.Matches(err, &AlreadyLockedError{}) {
+				return errs.Wrap(err, "Could not acquire lock")
+			}
+
+			select {
+			case <-timer.C:
+				return err
+			default:
+				time.Sleep(100 * time.Millisecond)
+			}
```

Comment:
This should work fine - just noting that you had thumbed-up my suggestion to use time.Time.After instead.

Reply:
Oh, I missed this... Using a timer requires more detailed usage: https://github.com/golang/go/issues/27169#issuecomment-415393233

I do think avoiding the channel by using time.Time.After is worthwhile here.

Reply:
Whoops, forgot to push the commit where I use `time.Now().After()`